### PR TITLE
Remove unused variable in Tuning test 

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: core/unit_test/tools/TestTuning.cpp [hunk 1]
```diff
@@ -53,8 +53,6 @@
 
 static size_t expectedNumberOfContextVariables;
 static int64_t expectedContextVariableValue;
-static std::unordered_map<size_t, Kokkos::Tools::Experimental::SetOrRange>
-    candidate_value_map;
```
Comment (dalg24) [inline core/unit_test/tools/TestTuning.cpp:57]: How comes we don't catch this in any of the CI builds?

**Review by DavidPoliakoff — APPROVED** [review]

File: core/unit_test/tools/TestTuning.cpp (hunk 1 above)
Comment (masterleinad) [reply core/unit_test/tools/TestTuning.cpp:57]: We only enable Tuning in the `OpenMPTarget` build where we don't use `-Werror`.

Comment (masterleinad) [reply core/unit_test/tools/TestTuning.cpp:57]: OK and CUDA-10.1-Clang-Tidy

Comment (dalg24) [reply core/unit_test/tools/TestTuning.cpp:57]: Why doesn't the `CUDA-10.1-Clang-Tidy` build catch it?

**Pull request merged**